### PR TITLE
feat(web): Add hold open toggle

### PR DESCRIPTION
**Pull request by jag3dagster**
- Add toggle to UI to allow a door to be held open once unlocked, useful for MLOs that have drive-thru windows but bad door tuning amongst other situations

- Note: This was authored by @Vicken52. He asked me to submit the PR as I already had the repro setup locally.

### CODE REVIEW

**Review by LukeWasTakenn — APPROVED** [review]

File: client/main.lua [hunk 1]
```diff
@@ -160,10 +160,12 @@ RegisterNetEvent('ox_doorlock:setState', function(id, state, source, data)
 		DoorSystemSetDoorState(double[1].hash, door.state, false, false)
 		DoorSystemSetDoorState(double[2].hash, door.state, false, false)
 
+		if door.holdOpen then DoorSystemSetHoldOpen(double[1].hash, door.state == 0); DoorSystemSetHoldOpen(double[2].hash, door.state == 0) end
```
Comment (thelindat) [inline client/main.lua]: Ugly one-liner.

```lua
if door.holdOpen then
    DoorSystemSetHoldOpen(double[1].hash, door.state == 0)
    DoorSystemSetHoldOpen(double[2].hash, door.state == 0)
end
```

File: web/src/layouts/settings/views/general/components/Switches.tsx [hunk 2]
```diff
@@ -45,6 +46,12 @@ const Switches: React.FC = () => {
           value={hideUi || false}
           toggle={() => toggleCheckbox('hideUi')}
         />
+        <TooltipSwitch
+          label="Hold Open"
+          infoCircle="Sets whether the targeting door should be held open"
```
Comment (thelindat) [inline web/src/layouts/settings/views/general/components/Switches.tsx]: Maybe more info than what the name describes?
> Sets whether the door should stay opened while unlocked.

**Review by thelindat — CHANGES_REQUESTED** [review]

**Comment by thelindat** [comment]
Should this maybe only be an option for automatic doors?

The option also isn't being saved (see encodeData on the server) or applied on load.

**Comment by jag3dagster** [comment]
> Should this maybe only be an option for automatic doors?

It works for regular doors as well. There are cases where someone may want to 'prop open a door' as an invite for people to come into a business or something.

**Pull request merged**